### PR TITLE
Added a Haskell-like Down interface

### PR DESCRIPTION
**Pull request by Synthetica9**

### CODE REVIEW

File: libs/prelude/Prelude/Interfaces.idr [hunk 1]
```diff
@@ -73,6 +73,21 @@ Eq Ordering where
...
+inverse : Ordering -> Ordering
+inverse LT = GT
+inverse EQ = EQ
+inverse GT = LT
+
+-- https://hackage.haskell.org/package/base-4.10.0.0/docs/Data-Ord.html#t:Down
+data Down a = MkDown a
+
+Eq a => Eq (Down a) where
+  Down x == Down y = x == y
```
Comment (msmorgan) [inline libs/prelude/Prelude/Interfaces.idr]: Needs to be `(Down x) == (Down y)`.

Comment (msmorgan) [inline libs/prelude/Prelude/Interfaces.idr]: Should be `(Down x) == (Down y)`

Comment (msmorgan) [inline libs/prelude/Prelude/Interfaces.idr]: Should be `(Down x) == (Down y)`.

**Comment by ahmadsalim** [comment]
Can you please provide a description of what this does?

File: libs/prelude/Prelude/Interfaces.idr (hunk 1 above)
Comment (Synthetica9) [reply libs/prelude/Prelude/Interfaces.idr]: Fixed.

**Comment by Synthetica9** [comment]
@ahmadsalim It allows you tocompare on the inverse of an ordering. This is useful for (for example) inversing sorts.

For example: 

```
> MkDown 3 < MkDown 2
True : Bool
```
It also adds the `inverse` function, which is basically `not` on orderings. (`GT` -> `LT`, `EQ` -> `EQ`, `LT` -> `GT`)

**Comment by ahmadsalim** [comment]
@Synthetica9 Thanks for the PR! I think this belongs more to `contrib` than `prelude`.

If you want it in `prelude` then there is a more strict review requirements, and you should send a mailing list update, etc.

**Comment by ahmadsalim** [comment]
I closed the PR, you are welcome to add it to contrib if you prefer instead with a new PR.

**Pull request closed without merging**